Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -1276,12 +1274,18 @@ void MediaPlayerPrivateGStreamerBase::handleProtectionStructure(const GstStructu
         return;
     }
 
-    String keySystemUUIDString = keySystemUUID ? keySystemUUID.get() : "(unspecified)";
     InitData initData(mappedInitData.data(), mappedInitData.size());
-    GST_TRACE("init data encountered for %s of size %" G_GSIZE_FORMAT " with MD5 %s", keySystemUUIDString.utf8().data(), mappedInitData.size(), GStreamerEMEUtilities::initDataMD5(initData).utf8().data());
-    GST_MEMDUMP("init data", mappedInitData.data(), mappedInitData.size());
+    initializationDataEncountered(initData, keySystemUUID ? keySystemUUID.get() : "(unspecified)");
+}
+
+void MediaPlayerPrivateGStreamerBase::initializationDataEncountered(const InitData& initData, const String& keySystemUUID)
+{
+    ASSERT(isMainThread());
+
+    GST_TRACE("init data encountered for %s of size %" G_GSIZE_FORMAT " with MD5 %s", keySystemUUID.utf8().data(), initData.sizeInBytes(), GStreamerEMEUtilities::initDataMD5(initData).utf8().data());
+    GST_MEMDUMP("init data", initData.characters8(), initData.sizeInBytes());
 
-    RunLoop::main().dispatch([weakThis = m_weakPtrFactory.createWeakPtr(*this), keySystemUUID = keySystemUUIDString, initData] {
+    RunLoop::main().dispatch([weakThis = m_weakPtrFactory.createWeakPtr(*this), keySystemUUID, initData] {
```

Comment:
We can probably use the helper method createWeakPtr instead of the factory soup here.

Reply:
Well, this is also outdated, but you made me look and it seems we don't need the WeakPtrFactory anymore, removing.